### PR TITLE
search: improve facet design.

### PR DESCRIPTION
**Pull request by zannkukai**
Improve the basic facet design :
  * facet entry count right aligned to the facet box.
  * text and counter is clickable to filter the search.
  * improves general design.

Co-authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## How to test?

image

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

**Review by iGormilhit — APPROVED** [review]

**Review by sebastiendeleze — APPROVED** [review]
OK for me but I noticed that the label of checkbox is blue, is it intentional? In your screenshot it's not the case.

**Comment by zannkukai** [comment]
@sebastiendeleze 
You are right ! It was because the label text is HTML rendered as a link. Now, I change the link style into facet to use the `$body-color` bootstrap variable color.
So now, the facet entry is colored as dark gray (#212529) except on hover state (color used is `$link-color` -- blue)

File: projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+.bucket-group {
+
+  .bucket-item {
+    font-size: 0.9rem;
+
+    .bucket-data {
+
+      .form-check-label {
+        color: $body-color;
+        padding: 0 0.5rem 0 0.3rem;
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss:31]: I would be great to use Bootstrap variables or css functions.

File: projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          text-decoration: underline;
+        }
+        .bucket-count {
+          font-weight: bold;
+        }
+      }
+
+      .bucket-count {
+        font-size: 0.8rem;
+        padding-left: 0.5rem;
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss:47]: same here

**Review by jma — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss (hunk 1 above)
Comment (zannkukai) [reply projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss]: done

File: projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss (hunk 2 above)
Comment (zannkukai) [reply projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.scss]: done

**Pull request merged**